### PR TITLE
Port Indirect Tools and Simulation

### LINKED ISSUE #25855: Port Indirect Simulation and Indirect Tools
### Description
Port the Indirect Tools and Simulation interface.

### Platforms affected
All

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR ports the remaining Indirect interfaces which don't have embedded plots, Indirect Tools and Simulations.

**To test:**
1. `Interfaces`->`Indirect`->`Simulations`
2. Load the data below for the relevant tabs
3. Click `Run` and wait.
4. Click `Plot` or `Plot Result` in the output options to make sure external plots appear.

1. `Interfaces`->`Indirect`->`Tools`->`Transmission` tab
2. Chemical Formula: `H2-O`
3. Mass density: 1.0
4. Click `Run`. No errors

Further work will need to be done for `LoadILL` to work completely in the future, but I consider this low priority for the next release.

**Data Files**
**MolDyn:**
Input File: [NaF_DISF.zip](https://github.com/mantidproject/mantid/files/2672564/NaF_DISF.zip)
FunctionNames: `Sqw-Na`

**Sassena:**
No data could be found as this is a largely unused interface. The plotting however does work.

**DensitiesOfStates:**
Input File: [csd_benzen11_PhonDisp.zip](https://github.com/mantidproject/mantid/files/2675281/csd_benzen11_PhonDisp.zip)

Fixes #25855

*This does not require release notes* because **the release notes for indirect will be updated when more of their useful interfaces are ported.**

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: qt/scientific_interfaces/Indirect/IndirectDiffractionReduction.cpp [hunk 1]
```diff
@@ -203,6 +217,7 @@ void IndirectDiffractionReduction::plotResults() {
   QString pyInput = "from mantidplot import plotSpectrum, plot2D\n";
```
Comment (Pasarus) [inline qt/scientific_interfaces/Indirect/IndirectDiffractionReduction.cpp]: Seems confusing that this is declared outside of `#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)` when it is only used inside of that macro (from what I can tell).

File: qt/scientific_interfaces/Indirect/IndirectDiffractionReduction.cpp [hunk 2]
```diff
@@ -224,6 +244,10 @@ void IndirectDiffractionReduction::plotResults() {
         showInformationBox(QString::fromStdString(
             "Workspace '" + it + "' not found\nUnable to plot workspace"));
     }
+#else
+    using MantidQt::Widgets::MplCpp::pcolormesh;
+    pcolormesh(toQStringList(m_plotWorkspaces));
+#endif
   }
 
   runPythonCode(pyInput);
```
Comment (Pasarus) [inline qt/scientific_interfaces/Indirect/IndirectDiffractionReduction.cpp]: You don't need this anymore, right? at least outside of a `#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)`

File: qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp [hunk 3]
```diff
@@ -124,12 +186,25 @@ void IndirectLoadILL::run() {
...
+    emit showMessageBox("IN16B is currently the only instrument supported in "
+                        "LoadILL on Mantid Workbench.");
+#endif
   }
-  runPythonScript(pyInput);
 
   setRunIsRunning(false);
 }
 
+void IndirectLoadILL::loadILLData(std::string const &filename,
```
Comment (Pasarus) [inline qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp]: Could this function go in the anonymous namespace alongside the other none-algorithm object changing functions?

File: qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp [hunk 4]
```diff
@@ -124,12 +186,25 @@ void IndirectLoadILL::run() {
                "'"
                ",'" +
                plot + "'," + save + ")";
+    runPythonScript(pyInput);
+#else
+    emit showMessageBox("IN16B is currently the only instrument supported in "
```
Comment (Pasarus) [inline qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp]: Do you know why we don't support anything but IN16B, wouldn't there be legacy data to support?

**Review by Pasarus — CHANGES_REQUESTED** [review]
Looking good so far a few comments ~~and a bit of an issue with the Indirect Tools page.~~

I had this setup:
image

But I did get an error which was:
```
Error in execution of algorithm SetSampleMaterial:
MaterialBuilder::setFormula() - Unable to parse chemical formula: Failed to find an atom with symbol=0 and a=0
Error in execution of algorithm IndirectTransmission:
MaterialBuilder::setFormula() - Unable to parse chemical formula: Failed to find an atom with symbol=0 and a=0
  at line 121 in '/home/sam/Repos/mantid1/Framework/PythonInterface/plugins/algorithms/IndirectTransmission.py'
  caused by line 1129 in '/home/sam/Repos/mantid1/Framework/PythonInterface/mantid/simpleapi.py'
Unknown error starting next batch algorithm
Got error from algorithm "IndirectTransmission"
Stopping batch algorithm because of execution error
```

**Comment by SpencerHowells** [comment]
@Pasarus @robertapplin looks like a typo to me. Water should be H2-O not with zero.

**Comment by Pasarus** [comment]
Ah yes! Indirect Tools page seems to be working fine, it bothered me that wasn't working.

File: qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp (hunk 4 above)
Comment (robertapplin) [reply qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp]: Getting workbench to support them will take some work. This is low priority and so can be implemented later on.

Comment (Pasarus) [reply qt/scientific_interfaces/Indirect/IndirectLoadILL.cpp]: Cool :+1:

**Review by Pasarus — APPROVED** [review]
I am happy with this now. Looks great! Good Job :+1: :1st_place_medal:

**Pull request merged**